browser-extension/chrome/popup/App.tsx

```
@@ -0,0 +1,207 @@
+  useEffect(() => {
+    checkAuth()
+    loadDefaultSettings()
+    checkPendingAction()
+  }, [])
+
+  // Extract content when mode changes
+  useEffect(() => {
+    if (isLoggedIn && !pendingAction?.text) {
+      extractContent(extractionMode)
+    }
+  }, [extractionMode, isLoggedIn, pendingAction?.text, extractContent])
```

Biome (2.3.13)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)